libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -554,12 +575,16 @@ export class PackageExtractor {
         // Replicate the links to the virtual store. Note that if the package has not been hoisted by
         // PNPM, the package will not be resolvable from here.
         // Only apply this logic for packages that were actually installed under the common/temp folder.
-        if (pnpmInstallFolder && Path.isUnder(packageJsonFolderPath, pnpmInstallFolder)) {
+        if (targetSubspace) {
           try {
             // The PNPM virtual store links are created in this folder.  We will resolve the current package
             // from that location and collect any additional links encountered along the way.
             // TODO: This can be configured via NPMRC. We should support that.
-            const pnpmDotFolderPath: string = path.join(pnpmInstallFolder, 'node_modules', '.pnpm');
+            const pnpmDotFolderPath: string = path.join(
+              targetSubspace.pnpmInstallFolder!,
```

Comment:
Could we remove the "!" sign here? Instead, adding it to if-condition

```
if (targetSubspace && targetSubspace.pnpmInstallFolder) {
```

Reply:
yes